app/models/manager_refresh/inventory_collection.rb

```diff
@@ -146,6 +149,11 @@ def object_index_with_keys(keys, object)
       keys.map { |attribute| object.public_send(attribute).to_s }.join(stringify_joiner)
     end
 
+    def find_or_build(manager_uuid)
+      # FIXME: splat manager_ref
+      data_index[manager_uuid] || build({ manager_ref.first => manager_uuid })
```

Comment:
if you want to pass the uuid as a string, then I do

manager_uuids    = manager_uuid.split(stringify_joiner)
hash_uuid_by_ref = manager_ref.zip(manager_uuids).to_h

then if manager ref is [:hardware, :device_name], the uuid in string is "#{hardware_ref}__#{device_name}"